library/memory_buffer_alloc.c

```diff
@@ -581,20 +584,24 @@ void mbedtls_memory_buffer_alloc_init( unsigned char *buf, size_t len )
     mbedtls_platform_set_calloc_free( buffer_alloc_calloc, buffer_alloc_free );
 #endif
 
-    if( (size_t) buf % MBEDTLS_MEMORY_ALIGN_MULTIPLE )
+    if( len < sizeof( memory_header ) + MBEDTLS_MEMORY_ALIGN_MULTIPLE )
```

Comment:
I think this is the correct behaviour. The buffer provided to memory manage must be at least as high as the memory header. If this is not the case, the internal `buffer_alloc_ctx` will be zeroized before returning and further calls to, say `mbedtls_calloc()` will signal `allocation failed` by returning a `NULL` pointer.